manager/src/main/java/org/openremote/manager/energy/ForecastSolarService.java

```diff
@@ -87,7 +91,7 @@ protected static class Result {
     protected static final AtomicReference<ResteasyClient> resteasyClient = new AtomicReference<>();
     protected ResteasyWebTarget forecastSolarTarget;
     private String forecastSolarApiKey;
-    private final Map<String, ScheduledFuture<?>> calculationFutures = new HashMap<>();
+    private final HashMap<String, ElectricityProducerSolarAsset> electricityProducerSolarAssetMap = new HashMap<>();
```

Comment:
Can it remain a `Map` instead? That way you can more easily change the implemention in the future.

```suggestion
    private final Map<String, ElectricityProducerSolarAsset> electricityProducerSolarAssetMap = new HashMap<>();
```